CV Backend/src/main/java/com/qa/demo/domain/Account.java

```diff
@@ -0,0 +1,115 @@
+package com.qa.demo.domain;
+
+import org.bson.types.Binary;
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotNull;
+
+@Document
+public class Account {
+    @Id
+    private ObjectId _id;
+
+    private String firstName;
+    private String lastName;
+    @Email
+    private String email;
+    @NotNull
+    private String password;
+
+    //private Role userRole;
+    @Field
+    private String userRole;
+    private Binary file;
+
+
+
+
+    public Account(){this.userRole="user";}
+
+    public Account(String firstName, String lastName) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        //this.userRole = RoleManager.getRoleList().giveRole("user");
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+    public Binary getFile() {
+        return file;
+    }
+
+    public void setFile(Binary file) {
+        this.file = file;
+    }
+
+    public String get_id() {
+        return _id.toHexString();
+    }
+
+    public void set_id(ObjectId _id) {
+        this._id = _id;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+    
+    public void updateFields(Account inAccount) {
+    	if (inAccount.getEmail() != null || !inAccount.getEmail().equalsIgnoreCase("")) {
```

Comment:
in these three if statements you're checking if it's not null, create a helper class that would have a method to which you pass a value and it checks whether it's not equal to null and returns a boolean for you

you also use this .equalsIgnoreCase("") repeatedly where you could have a helper method as well for doing this check for you and returning a boolean back